LEAF_Request_Portal/sources/Workflow.php

```diff
@@ -103,15 +103,34 @@ public function deleteStep($stepID)
     										WHERE nextStepID = :stepID', $vars);
         $res = $this->db->prepared_query('DELETE FROM workflow_steps
                                             WHERE stepID = :stepID', $vars);
-                                            
+
         $this->dataActionLogger->logAction(\DataActions::DELETE, \LoggableTypes::WORKFLOW_STEP, [
             new LogItem("workflow_steps", "stepID", $stepID),
             new LogItem("workflow_steps", "workflowID", $workflowID)
-        ]);  
+        ]);
 
         return 1;
     }
 
+    public function getStep(int $stepID): array
+    {
+        $vars = array(':stepID' => $stepID);
+        $workflowStepsRes = $this->db->prepared_query('SELECT 
```

Comment:
The query itself should be pulled out and assigned to a variable which is used here.